docs/content/tutorial/step_03.ngdoc

```diff
@@ -1,223 +1,279 @@
+DOM sub-tree constructed using the provided `template` and managed by an instance of the specified
+controller.
 
-In Step 2, we learned how to write and run unit tests. Unit tests are perfect for testing
-controllers and other components of our application written in JavaScript, but they can't easily
-test DOM manipulation or the wiring of our application. For these, an end-to-end test is a much
-better choice.
+But wait, where did that `$ctrl` come from and what does it refer to ?
 
-The search feature was fully implemented via templates and data-binding, so we'll write our first
-end-to-end test, to verify that the feature works.
+For reasons already mentioned (and for other reasons that are out of the scope of this tutorial), it
+is considered a good practice to avoid using the scope directly. We can (and should) use our
+controller instance; i.e. assign our data and methods on properties of our controller (the "`this`"
+inside the controller constructor), instead of directly to the scope.
 
-__`test/e2e/scenarios.js`:__
+From the template, we can refer to our controller instance using an alias. This way, the context of
+evaluation for our expressions is even more clear. By default, components use `$ctrl` as the
+controller alias, but we can override it, should the need arise. (It is a very uncommon need though,
+so we will be sticking with `$ctrl` most of the time.)
 
-```js
-describe('PhoneCat App', function() {
+There are more options available, so make sure you check out the
+{@link ng.$compileProvider#component API Reference}, before using `.component()` in your own
+applications.
 
-  describe('Phone list view', function() {
 
-    beforeEach(function() {
-      browser.get('app/index.html');
-    });
+## Using Components
 
+Now that we know how to create components, let's refactor the HTML page to make use of our newly
+acquired skill.
 
-    it('should filter the phone list as a user types into the search box', function() {
+<br />
+**`app/index.html`:**
 
-      var phoneList = element.all(by.repeater('phone in phones'));
-      var query = element(by.model('query'));
+```html
+<html ng-app="phonecatApp">
+<head>
+  ...
+  <script src="bower_components/angular/angular.js"></script>
+  <script src="app.js"></script>
+  <script src="phone-list.component.js"></script>
+</head>
+<body>
+
+  <!-- Use a custom component to render a list of phones -->
+  <phone-list></phone-list>
+
+</body>
+</html>
+```
 
-      expect(phoneList.count()).toBe(3);
+<br />
+**`app/app.js`:**
 
-      query.sendKeys('nexus');
-      expect(phoneList.count()).toBe(1);
+```js
+// Define the `phonecatApp` module
+angular.module('phonecatApp', []);
+```
 
-      query.clear();
-      query.sendKeys('motorola');
-      expect(phoneList.count()).toBe(2);
-    });
+<br />
+**`app/phone-list.component.js`:**
+
+```js
+// Register `phoneList` component, along with its associated controller and template
+angular.
+  module('phonecatApp').
```

Comment:
Ohhh! It's so hard to decide...

I am generally not in favour of reopening anything. Especially old milk bottles!

So I think I would rather go with one module per file:

* I don't believe that there is not so much of a performance penalty, because there is a similar cost to looking up the module when reopening.
* I believe that having small modules that explicitly declare their dependencies makes them a bit more self documenting.
* My feeling is that if things are so tightly bound that they should be in a single module then they should be in the same file??

If you do feel strongly @gkalpak  then don't do it. But if you are not bothered then I would be delighted to see this changed :-)